infrastructure/transform/main.tf

```diff
@@ -57,7 +57,7 @@ resource "azurerm_role_assignment" "adf_can_create_clusters" {
 }
 
 resource "azurerm_data_factory" "adf" {
-  name                = "${var.naming_prefix}-adf"
+  name                = replace("adf-${var.naming_suffix}", "_", "")
```

Comment:
I'm thinking perhaps we do away with `_` altogether and stipulate this in the `config.schema`, or we replace with `""`? There's a bit too much variation to account for with the current approach